pkg/logs/input/kubernetes/launcher.go

```diff
@@ -225,14 +226,27 @@ func (l *Launcher) getSourceName(pod *kubelet.Pod, container kubelet.ContainerSt
 // getPath returns a wildcard matching with any logs file of container in pod.
 func (l *Launcher) getPath(basePath string, pod *kubelet.Pod, container kubelet.ContainerStatus) string {
 	// the pattern for container logs is different depending on the version of Kubernetes
-	// so we need to try both format,
-	// until v1.13 it was `/var/log/pods/{pod_uid}/{container_name}/{n}.log`,
+	// so we need to try three possbile formats
+	// until v1.9 it was `/var/log/pods/{pod_uid}/{container_name_n}.log`,
+	// v.1.10 to v1.13 it was `/var/log/pods/{pod_uid}/{container_name}/{n}.log`,
 	// since v1.14 it is `/var/log/pods/{pod_namespace}_{pod_name}_{pod_uid}/{container_name}/{n}.log`.
 	// see: https://github.com/kubernetes/kubernetes/pull/74441 for more information.
 	oldDirectory := filepath.Join(basePath, l.getPodDirectoryUntil1_13(pod))
 	if _, err := os.Stat(oldDirectory); err == nil {
-		return filepath.Join(oldDirectory, container.Name, anyLogFile)
+		v110Dir := filepath.Join(oldDirectory, container.Name)
+		if _, err := os.Stat(v110Dir); err == nil {
```

Comment:
Does it make sense to log this error because it can be due to a different cause not just "file not present".

Reply:
makes sense, updating